### ISSUE TITLE
Add platforms support

### ISSUE THREAD

**Issue by vania-pooh** [aerokube/selenoid#672]
Should be able to match by platform name.

**Comment by manelpb** [comment]
Any chance we can get this feature anytime soon?

**Comment by vania-pooh** [comment]
@manelpb what is your use case?

**Comment by manelpb** [comment]
i have 2 selenoids running in Mac and Windows and I'd like to select chrome based on platform.

**Comment by vania-pooh** [comment]
@manelpb currently the most problematic thing is how to specify platform in `browsers.json` configuration. Where would you expect to have this setting?

**Comment by manelpb** [comment]
Yeah... i was looking into the code and came up with this json structure. What do you think? 

```
{
	"chrome": {
		"default": "latest",		
		"versions": [
			{
				"version": "latest",
				"platform": "linux",
				"image": "selenoid/vnc:chrome_78.0",
				"port": "4444",
				"enableVideo": true,
				"tmpfs": {
					"/tmp": "size=512m"
				}
			},
			{				
				"version": "latest",
				"platform": "mac",
				"image": "/usr/local/bin/chromedriver"
			}
		]
	}
}
```

Although that would be a breaking change :/ 

Another option would pass a argument variable to the selenoid app and based on that, selenoid would reject or accept sessions depending on the platform it's running. 

example: 

`./selenoid -conf ./config/browsers.json -platform mac`

**Comment by aandryashin** [comment]
Could you please check this project:

https://github.com/aerokube/ggr

There is platform support and it can be run in front of any number of
selenoids, selenium nodes, appiums and so on...

ср, 20 нояб. 2019 г., 19:47 Emmanuel Alves <notifications@github.com>:

**Comment by vania-pooh** [comment]
After thinking a bit more - platform support is needed only when several servers with Selenoids are installed. So I think using Ggr is a better option that overcomplicating Selenoid. Not closing right now, but postponing this issue.

**Comment by martingrossmann** [comment]
Hi, I want to remember of this ticket, I'm also interested in this feature.

My usecase is as follows:
* We are hosting our GGR/Selenoid cluster on 2 powerful hardware servers
* We created some Docker containers with a running Windows VM inspired by your description https://github.com/aerokube/windows-images
* So we could have a the same browser with the same version as a default Selenoid image (Linux) and as a Windows image 
* Both types of images are running on both servers
* At the moment Selenoid cannot differ between these two types of images. As a workaround I can change my version string of my Windows images (e.g. 94w) but this makes the plaform cap needless.
* Up to now we also have some real Windows VMs with some browser configurations. For them the GGR platform attribute is fine.

**Comment by vania-pooh** [comment]
@martingrossmann at least this rare case can be fixed by launching two separate Selenoid instances listening on different ports on the same machine.

**Comment by martingrossmann** [comment]
@vania-pooh Yes, this would be acceptable. :-) Thank you.
